### PR TITLE
implement entity particles custom metadata

### PR DESCRIPTION
**Pull request by msrivastav13**
### What does this PR do?

This PR implements Entity Particles for custom metadata type feature documented here (https://releasenotes.docs.salesforce.com/en-us/summer20/release-notes/rn_forcecom_cmt_entity_particle.htm)

### What issues does this PR fix or reference?

We were using text fields for metadata relationships for city and state fields in the app. With Summer 20 we can map individual entities of a compound field like an address.

Earlier one had to use Text fields and put a raw text. This would mean if admins put a wrong API name by mistake the app could break! 

Screen Shot 2020-07-29 at 9 53 30 PM

## The PR fulfills these requirements:

[ ] Tests for the proposed changes have been added/updated.
No test code required
[ ] Code linting and formatting was performed.
yes

### Functionality Before

<insert gif and/or summary>

### Functionality After

<insert gif and/or summary>

### CODE REVIEW

File: es-base-code/main/default/objects/Customer_Fields__mdt/fields/Customer_City__c.field-meta.xml [hunk 1]
```diff
@@ -1,15 +1,17 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <CustomField xmlns="http://soap.sforce.com/2006/04/metadata">
     <fullName>Customer_City__c</fullName>
-    <description
-    >Used because Address compound fields aren't yet available as independent fields in entity relationships.</description>
     <externalId>false</externalId>
     <fieldManageability>SubscriberControlled</fieldManageability>
     <inlineHelpText
-    >Enter the API name of the field that holds the customer's city information.</inlineHelpText>
+    >Enter the API name of the field that holds the customer&apos;s city information.</inlineHelpText>
```
Comment (adityanaag3) [inline es-base-code/main/default/objects/Customer_Fields__mdt/fields/Customer_City__c.field-meta.xml]: I think we can remove the help text. None of the other fields which have the Field Def lookup have help texts.

File: es-base-code/main/default/objects/Customer_Fields__mdt/fields/Customer_State__c.field-meta.xml [hunk 2]
```diff
@@ -1,15 +1,17 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <CustomField xmlns="http://soap.sforce.com/2006/04/metadata">
     <fullName>Customer_State__c</fullName>
-    <description
-    >Used because Address compound fields aren't yet available as independent fields in entity relationships.</description>
     <externalId>false</externalId>
     <fieldManageability>SubscriberControlled</fieldManageability>
     <inlineHelpText
-    >Enter the API name of the field that holds the customer's state information.</inlineHelpText>
+    >Enter the API name of the field that holds the customer&apos;s state information.</inlineHelpText>
```
Comment (adityanaag3) [inline es-base-code/main/default/objects/Customer_Fields__mdt/fields/Customer_State__c.field-meta.xml]: Same as above

**Review by adityanaag3 — CHANGES_REQUESTED** [review]
This is great. Good catch. Requested minor changes for consistency

File: es-base-code/main/default/objects/Customer_Fields__mdt/fields/Customer_State__c.field-meta.xml (hunk 2 above)
Comment (msrivastav13) [reply es-base-code/main/default/objects/Customer_Fields__mdt/fields/Customer_State__c.field-meta.xml]: done!

File: es-base-code/main/default/objects/Customer_Fields__mdt/fields/Customer_City__c.field-meta.xml (hunk 1 above)
Comment (msrivastav13) [reply es-base-code/main/default/objects/Customer_Fields__mdt/fields/Customer_City__c.field-meta.xml]: done!

**Review by pozil — APPROVED** [review]
Great addition. Thx @msrivastav13

**Pull request merged**